fendermint/app/src/cmd/run.rs

```diff
@@ -27,7 +29,8 @@ async fn run(settings: Settings) -> anyhow::Result<()> {
     );
     let interpreter = SignedMessageInterpreter::new(interpreter);
     let interpreter = ChainMessageInterpreter::new(interpreter);
-    let interpreter = BytesMessageInterpreter::new(interpreter);
+    let interpreter =
+        BytesMessageInterpreter::new(interpreter, ProposalPrepareMode::AppendOnly, false);
```

Comment:
What is the rationale for not rejecting a malformed proposal here (if any, as maybe is just a placeholder value for now)?

Reply:
I included the [considerations](https://github.com/consensus-shipyard/fendermint/pull/185/files#diff-5eac90311bc4bcf59a295948bfe9b9c82b725532cff3ce4036d92a8654052b85R118-R126) in the code as comments where this flag is used. 

The reason I thought we might not want to reject them for now is because 
1. They should not happen as our nodes are honest. 
2. If they happen, we have a bug with a submitted transaction, and this way the error will show up in the transaction results which the RPC client will get and print, so the test will fail with an error, rather than a timeout. Whereas if we reject the proposal, CometBFT itself will most likely die as a result because our honest nodes will have proposed something that they subsequently reject themselves.  And with `AppendOnly` we are not going to inspect the bytes before adding them to the proposal, as it was pointed out as wasted effort by @cryptoAtwill . That will have to change if we want to limit gas, though. 
3. If we admitted a malformed transaction to our mempool, inspected it (by using `PassThrough` instead of `AppendOnly`), and not included it in our proposal, it will still stay in the mempool, resulting in a memory leak in CometBFT if we keep piling up such transactions. This would be another bug that is potentially hard to figure out, apart from flooding the logs with warnings by trying to propose the same transaction every 2 seconds. 

If all our nodes are not honest and some are proposing faulty transactions then this would be a stupid way to attack instead of just not proposing any block to cause a timeout. If it happened, our only chance to weed out such actors is to admit the transaction and penalize the miner. If that happened due to a bug, we'd risk consensus failure, but at least we'd see what's happening, and by coding the penalty mechanism we'd discourage anyone from intentionally doing this, and to implement these checks in `check_tx`. 

So, overall, I thought that a malformed TX here is most likely a bug that is easiest to detect by letting it through.

Reply:
That makes sense! Assuming our implementation honest saving ourselves from nasty bugs is the way to go, thanks!